vllm/compilation/decorators.py

```diff
@@ -0,0 +1,83 @@
+from typing import List, Optional, Union
+
+import torch
+
+import vllm.envs as envs
+from vllm.attention import AttentionMetadata
+from vllm.compilation.levels import CompilationLevel
+from vllm.compilation.wrapper import TorchCompileWrapperWithCustomDispatcher
+from vllm.sequence import IntermediateTensors
+
+
+def support_compile_llama_style(cls: type):
+    """
+    A decorator to add support for compiling the forward method of a class.
+    If a module's **forward signature** is compatible with llama, this 
+    decorator can be used to enable the compilation of the forward method.
+    """
+
+    if envs.VLLM_TORCH_COMPILE_LEVEL == CompilationLevel.NO_COMPILATION:
+        return cls
+
+    # take care of method resolution order
+    # make sure super().__init__ is called on the base class
+    #  other than TorchCompileWrapperWithCustomDispatcher
+    cls.__bases__ = cls.__bases__ + (TorchCompileWrapperWithCustomDispatcher, )
+
+    old_init = cls.__init__
+
+    def __init__(self, *args, **kwargs):
+        old_init(self, *args, **kwargs)
+        self._use_torch_compile = \
+            envs.VLLM_TORCH_COMPILE_LEVEL > CompilationLevel.NO_COMPILATION
+        if self._use_torch_compile:
+            TorchCompileWrapperWithCustomDispatcher.__init__(self)
+
+    cls.__init__ = __init__
```

Comment:
Nit: Instead of manually editing the class properties, you could define a new class that inherits from `cls`:
```python
def support_compile_llama_style(cls: type):
  class Decorated(cls):
      def __init__(...):
        super().__init__(...)
     ...

  return Decorated
```

Reply:
this might break some code and causes confusion, people might expect the class name is something like `LlamaModel` , rather than a `Decorated` class.